src/ui/components/Switch/Switch.scss

```diff
@@ -110,101 +110,101 @@ $installStripeColor2: #00c42e;
     }
   }
 
-  &.uninstalling {
-    input + label::before {
-      @include stripes(
-        $uninstallStripeColor1,
-        $uninstallStripeColor2,
-        $switchBackgroundOff);
-      // Reverse the stripe direction without
-      // impacting the animation speed.
-      transform: scaleX(-1);
-    }
-
-    input + label {
-      &::after {
-        transform: translateX($switchHandleGap);
+  // Disabled State
+  input:disabled + label,
+  input:checked:disabled + label {
+    cursor: not-allowed;
 
-        /* stylelint-disable max-nesting-depth */
-        [dir=rtl] & {
-          transform: translateX(-$switchHandleGap);
-        }
-        /* stylelint-enable */
-      }
+    &::after {
+      background: $switchHandleInactive;
+    }
 
-      &:hover::after {
-        box-shadow: none;
-      }
+    &:hover::after {
+      box-shadow: none;
     }
   }
 
-  &.downloading {
-    overflow: hidden;
-    position: relative;
+  input:checked + label {
+    border-color: $switchStrokeOn;
 
-    input + label {
-      border: 1px solid $switchStrokeOff;
-      overflow: hidden;
+    &::before {
+      background: $switchBackgroundOn;
+    }
 
-      &::after {
-        background: $switchHandleActive;
-        transform: translateX($switchHandleActivePosition);
+    &::after {
+      transform: translateX($switchHandleActivePosition);
 
-        /* stylelint-disable max-nesting-depth */
-        [dir=rtl] & {
-          transform: translateX(-$switchHandleActivePosition);
-        }
-        /* stylelint-enable */
+      [dir=rtl] & {
+        transform: translateX(-$switchHandleActivePosition);
       }
     }
   }
 
-  &.installing {
-    input + label {
-      border-color: $switchStrokeOn;
+  &.Switch--success {
+    input + label::before {
+      background: $switchBackgroundOn url('./tick.svg') no-repeat 35% 50%;
 
-      &::before {
-        @include stripes(
-          $installStripeColor1,
-          $installStripeColor2,
-          $switchBackgroundOn);
+      [dir=rtl] & {
```

Comment:
I didn't know about this kind of parent/child inversion with the ampersand, seems handy